src/Chapter3.hs

```diff
@@ -479,6 +527,54 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living people inside
 -}
+-- our city
+data City
+  = City
+   { cityCastle :: Castle
+   , cityBuild  :: Building
+   , cityHouses :: [House]
+   }
+
+-- our castle
+data Castle
+    = None    -- dont have castle, but I cannot see what is None from your Chapter
+    | OnlyCastle -- one castle
+    | CastleWithWalls -- castle with wall
```

Comment:
The task tells that the castle should have a name. In this case, it can be a `String`. Sorry, if it wasn't clear before. We improved this task wording in the latest version to clarify this moment.
Could you guess, what functions would need patching in case of the type modifications?

Reply:
I forgot about enums. This data just enum with my values. It's not the same as Nothing from Maybe.
I understood now.